File: 11.container-with-most-water.java
/*
 * @lc app=leetcode id=11 lang=java
 *
 * [11] Container With Most Water
 */

// @lc code=start
class Solution {
    public int maxArea(int[] height) {
        
        int left = 0;
        int right = height.length - 1;
        int maxAreaValue = 0;

        while (left < right) {

            maxAreaValue = Math.max(maxAreaValue, (right - left) * Math.min(height[left], height[right]));

            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }

        return maxAreaValue;
    }
}
// @lc code=end

File: 11.container-with-most-water.py
#
# @lc app=leetcode id=11 lang=python3
#
# [11] Container With Most Water
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        left = 0
        right = len(height) - 1
        max_area = 0

        while left < right:
            max_area = max(max_area, (right - left) * min(height[right], height[left]))
            
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        
        return max_area
# @lc code=end

